tests/test_operators.py

```diff
@@ -80,16 +83,170 @@ def test_flatten_instances(self):
         test_operator(operator=FlattenInstances(sep="..."), inputs=inputs, targets=targets, tester=self)
 
     def test_filter_by_values(self):
+        inputs = [{"a": 1, "b": 2}, {"a": 2, "b": 3}, {"a": 1, "b": 3}]
+
+        targets = [
+            {"a": 1, "b": 3},
+        ]
+
+        test_operator(
+            operator=FilterByValues(required_values={"a": 1, "b": 3}), inputs=inputs, targets=targets, tester=self
+        )
+
+        with self.assertRaises(ValueError) as cm:
+            test_operator(
+                operator=FilterByValues(required_values={"c": "5"}), inputs=inputs, targets=targets, tester=self
+            )
+        self.assertEqual(
+            str(cm.exception),
+            "Required filter field ('c') in FilterByValues is not found in {'a': 1, 'b': 2}",
+        )
+
+    def test_filter_by_list_of_values(self):
         inputs = [
             {"a": 1, "b": 2},
             {"a": 2, "b": 3},
+            {"a": 3, "b": 4},
         ]
 
         targets = [
-            {"a": 1, "b": 2},
+            {"a": 2, "b": 3},
+            {"a": 3, "b": 4},
+        ]
+
+        test_operator(
+            operator=FilterByListsOfValues(required_values={"b": ["3", "4"]}),
+            inputs=inputs,
+            targets=targets,
+            tester=self,
+        )
+
+        with self.assertRaises(ValueError) as cm:
+            test_operator(
+                operator=FilterByListsOfValues(required_values={"b": "5"}), inputs=inputs, targets=targets, tester=self
+            )
+        self.assertEqual(str(cm.exception), "The filter for key ('b') in FilterByListsOfValues is not a list but '5'")
+
+        with self.assertRaises(ValueError) as cm:
+            test_operator(
+                operator=FilterByListsOfValues(required_values={"c": ["5"]}),
+                inputs=inputs,
+                targets=targets,
+                tester=self,
+            )
+        self.assertEqual(
+            str(cm.exception), "Required filter field ('c') in FilterByListsOfValues is not found in {'a': 1, 'b': 2}"
+        )
+
+    def test_intersect(self):
+        inputs = [
+            {"label": ["a", "b"]},
+            {"label": ["a", "c", "d"]},
+            {"label": ["a", "b", "f"]},
+        ]
+
+        targets = [
+            {"label": ["b"]},
+            {"label": []},
+            {"label": ["b", "f"]},
         ]
 
-        test_operator(operator=FilterByValues(values={"a": 1}), inputs=inputs, targets=targets, tester=self)
+        test_operator(
```

Comment:
suggestion (not crucial): add `test_operator_and_excpect_exception`, or similar. since this seems to repeat several times.

Reply:
Good.  I'll fix this in the next operator I write. I want the operators to be available to Ilya.